source/guides/kernel/firmware.rst

```diff
@@ -0,0 +1,77 @@
+.. _firmware:
+
+Firmware
+########
+
+This guide shows how |CL-ATTR| handles firmware or microcode loading.
+
+.. contents::
+   :local:
+   :depth: 1
+
+Overview 
+********
+
+Many devices and system components require firmware or microcode, software
+that runs directly on the device, to function correctly. Because firmware
+loading requires privileged hardware access, the kernel is involved in the
+process.
+
+Firmware does not typically come with source code. Instead, firmware is
+provided as binary blobs which are licensed for free or non-free use.
+
+In |CL| firmware is loaded during device initialization which typically
+happens at boot time. 
+
+
+Included firmware
+*****************
+
+The Linux kernel sources contain firmware binaries that are licensed to allow
```

Comment:
Maybe clarify that it's not the kernel sources containing, but rather the kernel project distributes firmware binaries.